src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -405,6 +405,7 @@ private function validateAndDenormalize(string $currentClass, string $attribute,
 
         $expectedTypes = [];
         $isUnionType = \count($types) > 1;
+        $extraAttributes = [];
```

Comment:
In my opinion, `extraAttributes` should not be merged between different types.
I think the best would to save the first `ExtraAttributesException` thrown and throw it again if not null below.

Reply:
I am not against it. The purpose of this PR to allow deserialization, when the first union does not contain a field provided in the serialized data, but one of the other union type is a perfect match.

Reply:
We could even the list of extra attributes per type in the exception message.

Reply:
I modified the code according to the suggestion of @NorthBlue333.